src/convergence/Convergence.jl

```diff
@@ -0,0 +1,59 @@
+module Convergence
+
+using Distributions
+using Flux
+using LinearAlgebra
+using ..CounterfactualExplanations
+using ..Generators
+using ..Models
+
+include("decision_threshold.jl")
+include("generator_conditions.jl")
+include("invalidation_rate.jl")
+include("max_iter.jl")
+
+"""
+    convergence_catalogue
+
+A dictionary containing all convergence criteria.
+"""
+const convergence_catalogue = Dict(
+    :decision_threshold => DecisionThresholdConvergence(),
+    :generator_conditions => GeneratorConditionsConvergence(),
+    :max_iter => MaxIterConvergence(),
+    :invalidation_rate => InvalidationRateConvergence(),
+)
+
+"""
+    get_convergence_type(convergence::AbstractConvergence)
+
+Returns the convergence object.
+"""
+function get_convergence_type(convergence::AbstractConvergence)
+    return convergence
+end
+
+"""
+    get_convergence_type(convergence::Symbol)
+
+Returns the convergence object from the dictionary of default convergence types.
+"""
+function get_convergence_type(convergence::Symbol)
+    return get(
+        convergence_catalogue,
+        convergence,
+        () -> error("Convergence criterion not recognized: $convergence."),
+    )
+end
```

Comment:
![image](https://github.com/JuliaTrustworthyAI/CounterfactualExplanations.jl/assets/67654024/02eef850-e99f-4f36-a924-96dd9dacc776)

Using this `get` method you introduce incompatibility with Julia 1.6 and lower. Is it necessary to use it?

Reply:
We're using neither tuples nor numbers here, so I guess it's fine?